src/codegen/arithmetic_translator.cpp

```diff
@@ -0,0 +1,88 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// arithmetic_translator.cpp
+//
+// Identification: src/codegen/arithmetic_translator.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "codegen/arithmetic_translator.h"
+
+#include "codegen/value_proxy.h"
+#include "expression/operator_expression.h"
+
+namespace peloton {
+namespace codegen {
+
+// Constructor
+ArithmeticTranslator::ArithmeticTranslator(
+    const expression::OperatorExpression &arithmetic,
+    CompilationContext &context)
+    : ExpressionTranslator(arithmetic, context), ctx_(context) {
```

Comment:
@avahuang1 Can we modify ExpressionTranslator to store the compilation context so that we do not have to add this to all the children of ExpressionTranslator?